### PR TITLE
[Tabs] Adds the Ripple behavior to Tabs.

### LINKED ISSUE #7379: [Ripple] Internal issue: b/132663279
This was filed as an internal issue. If you are a Googler, please visit [b/132663279](http://b/132663279) for more details.

---
#### Internal data
- Associated internal bug: [b/132663279](http://b/132663279)

### PR DESCRIPTION
**Pull request by yarneo**
## Related links
* Bug: Closes #7379 
* Ripple: [MDCRippleTouchController](https://github.com/material-components/material-components-ios/tree/develop/components/Ripple)

## Introduction
This PR integrates [`MDCRippleTouchController`](https://github.com/material-components/material-components-ios/blob/develop/components/Ripple/src/MDCRippleTouchController.h) into [`MDCTabBar`](https://github.com/material-components/material-components-ios/tree/develop/components/Tabs). This behavior is an opt in flag property added to MDCTabBar, `enableRippleBehavior`.  This improves the visual touch feedback for our users as MDCInkView didn't support our current motion guidelines. We are also marking any of the Ink APIs ToBeDeprecated as part of the [Deprecation Policy](https://github.com/material-components/material-components-ios/blob/cb6b8e9599e9018c545e0011a3ded82597988106/contributing/deprecation_policy.md#step-4-add-a-comment-to-the-api-indicating-that-it-will-be-deprecated).

## Videos
| Before | After |
| - | - |
|oldRipple|newRipple|

### CODE REVIEW

**Review by bryanoltman — APPROVED** [review]

File: components/Tabs/src/MDCTabBar.h [hunk 1]
```diff
@@ -90,8 +90,16 @@ IB_DESIGNABLE
...
+/**
+ Determines if an @c MDCTabBar should use the @c MDCRippleView behavior or not.
+ By setting this property to @c YES, @c MDCRippleView is used to provide the user visual
+ touch feedback, instead of the legacy @c MDCInkView.
+ @note Defaults to @c NO.
+ */
+@property(nonatomic, assign) BOOL enableRippleBehavior;
+
+/** Ripple color for taps on tab bar items. Default: Semi-transparent white. */
+@property(nonatomic, nonnull) UIColor *rippleColor UI_APPEARANCE_SELECTOR;
```
Comment (jverkoey) [inline components/Tabs/src/MDCTabBar.h]: Please do not add more `UI_APPEARANCE_SELECTOR` annotations to our APIs.

File: components/Tabs/src/MDCTabBar.m [hunk 2]
```diff
@@ -292,6 +299,18 @@ - (void)setInkColor:(UIColor *)inkColor {
   }
 }
 
+- (UIColor *)inkColor {
+  return _inkColor;
+}
+
+- (void)setRippleColor:(UIColor *)rippleColor {
+  if (_rippleColor != rippleColor && ![_rippleColor isEqual:rippleColor]) {
```
Comment (jverkoey) [inline components/Tabs/src/MDCTabBar.m]: Please update this to an early-exit clause, similar to the logic below in `setEnableRippleBehavior:`

```objc
if (_rippleColor == rippleColor || [_rippleColor isEqual:rippleColor]) {
  return;
}
```

File: components/Tabs/src/MDCTabBar.m [hunk 3]
```diff
@@ -78,6 +79,10 @@ static MDCItemBarAlignment MDCItemBarAlignmentForTabBarAlignment(
   return MDCItemBarAlignmentLeading;
 }
 
+static inline UIColor *kRippleColor() {
```
Comment (jverkoey) [inline components/Tabs/src/MDCTabBar.m]: Static methods do not have `k` prefixes; `k` prefixes are reserved for constants. `RippleColor` is the preferred name in this case.

File: components/Tabs/tests/unit/MDCTabBarRippleTests.m [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+      }
+    }
+  }
+  return arr;
+}
+
+/**
+ Test to confirm behavior of initializing a @c MDCTabBar without any customization.
+ */
+- (void)testDefaultTabBarBehaviorWithInkEnabled {
```
Comment (jverkoey) [inline components/Tabs/tests/unit/MDCTabBarRippleTests.m]: Similarly here as in https://github.com/material-components/material-components-ios/pull/7596#discussion_r293630962

File: components/Tabs/src/MDCTabBar.h [hunk 5]
```diff
@@ -90,8 +90,16 @@ IB_DESIGNABLE
  */
 @property(nonatomic, nonnull) UIColor *unselectedItemTintColor UI_APPEARANCE_SELECTOR;
 
-/** Ink color for taps on tab bar items. Default: Semi-transparent white. */
-@property(nonatomic, nonnull) UIColor *inkColor UI_APPEARANCE_SELECTOR;
+/**
+ Determines if an @c MDCTabBar should use the @c MDCRippleView behavior or not.
+ By setting this property to @c YES, @c MDCRippleView is used to provide the user visual
+ touch feedback, instead of the legacy @c MDCInkView.
+ @note Defaults to @c NO.
```
Comment (jverkoey) [inline components/Tabs/src/MDCTabBar.h]: Same here similarly to https://github.com/material-components/material-components-ios/pull/7596#discussion_r293630410

**Pull request merged**